packages/athena/src/client.ts

```diff
@@ -62,55 +72,35 @@ export class AthenaClient {
     Data extends object = object,
     Variables extends DefaultVariables = DefaultVariables
   >(
-    queryOperation: DocumentInput<Data, Variables>,
+    operation: DocumentInput<Data, Variables>,
     variables?: Variables
   ): Promise<OperationResult<Data>> {
-    const query = prepareQuery<Data, Variables>(queryOperation);
-
-    const result = this.makeRequest<Data, Variables>({
-      query,
-      variables,
-    });
+    const prepared = prepareOperation<Data, Variables>(operation, variables);
 
-    return result;
+    return this.makeRequest<Data, Variables>(prepared);
   }
 
   async mutate<
     Data extends object = object,
     Variables extends DefaultVariables = DefaultVariables
   >(
-    queryOperation: DocumentInput<Data, Variables>,
+    operation: DocumentInput<Data, Variables>,
     variables?: Variables
   ): Promise<OperationResult<Data>> {
-    const query = prepareQuery(queryOperation);
-
-    const result = await this.makeRequest<Data, Variables>({
-      query,
-      variables,
-    });
+    const prepared = prepareOperation<Data, Variables>(operation, variables);
 
-    return result;
+    return this.makeRequest<Data, Variables>(prepared);
   }
+
   private async makeRequest<
     Data extends object = object,
     Variables extends DefaultVariables = DefaultVariables
   >(request: GraphQLRequest<Data, Variables>): Promise<OperationResult<Data>> {
     let response: Response;
     const result: OperationResult<Data> = {};
 
-    const body = {
-      query: print(request.query),
-      variables: request.variables,
-    };
-
     try {
-      response = await this.fetch(this.url, {
-        method: 'POST',
-        headers: {
-          'Content-Type': 'application/json',
-        },
-        body: JSON.stringify(body),
-      });
+      response = await this.fetch(this.url, this.buildRequest(request));
```

Comment:
As mentioned above, I'd _generally_ prefer this was:

```suggestion
      response = await this.fetch(this.buildRequest(request));
```